synchronizer/batches.go

```diff
@@ -101,13 +105,25 @@ func (bs *BatchSynchronizer) Start(ctx context.Context) {
 	go bs.processUnresolvedBatches(ctx)
 	go bs.produceEvents(ctx)
 	go bs.handleReorgs(ctx)
+	go bs.startOffchainDataGapsDetection(ctx)
 }
 
 // Stop stops the synchronizer
 func (bs *BatchSynchronizer) Stop() {
 	close(bs.stop)
 }
 
+// Gaps returns the offchain data gaps
+func (bs *BatchSynchronizer) Gaps() map[uint64]uint64 {
+	bs.offchainDataGapsLock.Lock()
```

Comment:
Maybe do `defer unlock` straight away?

Reply:
There is only one return statement here, no need to defer execution in this case. Deferred execution consumes more resources so better to use it when it makes sense.